### PR TITLE
PR 6: Extract run_processor module from find_boundaries

### PR DESCRIPTION
**Pull request by ebanerjeeTT**
## Summary
Extracts workflow run processing into `modules/boundaries/run_processor.sh` per refactoring plan PR 6.

## Changes
- **Add** `modules/boundaries/run_processor.sh`: `process_workflow_runs()`, `is_commit_newer()`, pagination/filtering/cutoff via `job_matcher.sh`
- **Refactor** `find_boundaries.sh` to use `run_processor` instead of inline run-processing loop (~190 lines removed)
- **Add** `tests/modules/boundaries/run_processor_test.sh` (6 smoke tests)
- **Fix** `stop_search` propagation when failure limit exceeded

## Testing
All 97 lib + module tests pass.

proof it doesn't break auto-triage: https://github.com/tenstorrent/tt-metal/actions/runs/22200194468 

Note: One behavioural change. Now, if auto-triage fails to find a subjob for 50 consecutive runs, it will terminate. I intended this to be the behavior in a previous PR ([see pr 73](https://github.com/tenstorrent/tt-auto-triage/commit/6b304153fbbd409b32c494c3352b2e2bb0bbe129)) but I overlooked this edge case.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: .github/actions/auto-triage/auto_triage/modules/boundaries/run_processor.sh [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+    local page=1 processed=0
+    local last_successful_run="" last_successful_run_id="" last_successful_commit="" last_successful_job_url=""
+    local first_failing_run="" first_failing_run_id="" first_failing_commit="" first_failing_job_url=""
+    local most_recent_failure_run="" most_recent_failure_run_id="" most_recent_failure_commit="" most_recent_failure_job_url=""
+    local found_success=false stop_search=false consecutive_missing=0
+    local failure_only_count=0 exceeded_failure_limit=false
+    local subjob_runs_json='[]'
+
+    while true; do
```
Comment (ttmchiou) [inline .github/actions/auto-triage/auto_triage/modules/boundaries/run_processor.sh]: this inherently leaves with bad vibes when its while true;
any chance there is a guaranteed end condition?
